### DISCUSSION TITLE
Would it be possible to support a mutationKeyHashFn option?

### DISCUSSION THREAD

**Discussion by luke-layerhealth** [TanStack/query#7457] [Ideas]
My keys can contain `bigInt` which isn't JSON serializable. I don't have direct control over the keys since I'm using [connect-query-es](https://github.com/connectrpc/connect-query-es). These keys are not compatible with the default `hashKey`. I can provide my own `hashKey` implementation using `queryKeyHashFn` but there is no similar option for mutation keys. If you're cool with it I'll provide a PR adding the `mutationKeyHashFn` following the same patterns and tests used by the `queryKeyHashFn` implementation.

@TkDodo tagging you since you've been quite quick responding to these previously and I'm willing to put up a PR.

**Comment by TkDodo** [comment]
sorry for the late reply; yeah I think it would be good to have that for consistency 👍

**Comment by lpettecrew** [comment]
Hi @luke-layerhealth did you manage to find a work-around? I'm also using bigints which is crashing on the hash function.

Edit:
Found a workaround by serialising the mutation keys for each mutation. Bit annoying, but as a workaround fine for now.
Would still be good to get mutationKeyHashFn in for consistency though.
